core/chains/evm/config/config.go

```diff
@@ -79,6 +72,15 @@ type ChainScopedOnlyConfig interface {
 type EVM interface {
 	BalanceMonitor() BalanceMonitor
 	Transactions() Transactions
+
+	AutoCreateKey() bool
+	BlockBackfillDepth() uint64
+	BlockBackfillSkip() bool
+	FinalityDepth() uint32
+	LogBackfillBatchSize() uint32
+	LogPollInterval() time.Duration
+	LogKeepBlocksDepth() uint32
+	NonceAutoSync() bool
```

Comment:
just a thought since you are also splitting things down based on services - nothing to change right now, just writing it down 😄 

eventually each chain config should probably be composed of service configs, since we would want to be able to run multiple chain types on a single CL node - but I can also see you are headed towards that direction with your changes 🎉